cli/src/Morphir/Web/DevelopApp.elm

```diff
@@ -1213,6 +1255,54 @@ viewHome model packageName packageDef =
 
                         Nothing ->
                             row [ width fill, spacing (Theme.smallSpacing model.theme), padding (Theme.smallPadding model.theme) ] [ text <| "This package contains " ++ String.fromInt numberOfModules ++ " modules." ]
+
+                -- View to display the ValueEditors for Custom Attributes when a node is selected
+                viewAttributeValues : NodeID -> Element Msg
+                viewAttributeValues node =
+                    let
+                        attributeToEditors : Element Msg
+                        attributeToEditors =
+                            model.customAttributes
+                                |> Dict.toList
+                                |> List.map
+                                    (\( attrId, attrDetail ) ->
+                                        let
+                                            irValue : Maybe (Value () ())
+                                            irValue =
+                                                attrDetail.data
+                                                    |> SDKDict.get node
+
+                                            nodeDetail : AttrValueDetail
+                                            nodeDetail =
+                                                { attrId = attrId, nodeId = node }
+
+                                            editorState : ValueEditor.EditorState
+                                            editorState =
+                                                model.attributeStates
+                                                    |> SDKDict.get nodeDetail
+                                                    |> Maybe.withDefault
+                                                        (ValueEditor.initEditorState
+                                                            (IR.fromDistribution attrDetail.iR)
+                                                            (Type.Reference () attrDetail.entryPoint [])
+                                                            irValue
```

Comment:
Using the `irValue` is duplicating the logic that you already have in the update function where you populate `model.attributeStates` from the data that came back from the server. You should instead just leave the editor uninitialized if the state cannot be found which can be done by passing a `Nothing` here.

Reply:
Redundant code was removed from the ServerGetAttributeResponse Message since this message is send only during initialization. At this point, the selectedModule state is not initialized and therefore lookup cannot be done. So we are setting the  editorStates in the viewHome function using the Attribute Message